CHANGELOG.md

```diff
@@ -1,5 +1,61 @@
 # Changelog
 
+## Version 15
+
+### 15.0.0
+
+- **Breaking changes**:
+  - `express-fileupload` and `compression` become optional peer dependencies;
+  - Methods `createServer()` and `attachRouting()` become async;
+  - Method `testEndpoint()` requires an additional argument;
+  - Method `createLogger()` removed as redundant;
+  - Read the migration guide below.
+- Features:
+  - `winston` is now optional: supporting any logger having `debug()`, `warn()`, `info()` and `error()` methods;
+  - Introducing module augmentation approach for setting the type of chosen logger;
+  - Supporting both `jest` and `vitest` frameworks for `testEndpoint()`.
+- How to migrate while maintaining previous functionality and behavior:
+  - If you're going to continue using `winston`:
+    - Near your `const config = createConfig(...)` add the module augmentation statement (see example below).
+  - If you have `upload` option enabled in your config:
+    - Install `express-fileupload` and `@types/express-fileupload` packages;
+  - If you have `compression` option enabled in your config:
+    - Install `compression` and `@types/compression` packages;
+  - If you're using `createLogger()` method:
+    - Place its argument as a value of `logger` property supplied to `createConfig()`.
+  - If you're using the entities returned from `createServer()` or `attachRouting()` methods:
+    - Add `await` before calling them: `const {...} = await createServer(...)`.
+    - If you can not use `await` (on the top level of CommonJS):
+      - Wrap your code with async IIFE or use `.then()` (see example below).
+  - If you're using `testEndpoint()` method:
+    - Specify either `mockFn: jest.fn` or `mockFn: vi.fn` within its object argument.
+
+```typescript
+import type { Logger } from "winston";
+import { createConfig, createServer, testEndpoint } from "express-zod-api";
+
+// The configuration remains the same
+const config = createConfig({
+  logger: { level: "debug", color: true },
+});
+
+// Set the type of logger used near your configuration
+declare module "express-zod-api" {
+  interface LoggerOverrides extends Logger {}
+}
+
+// await is only needed when you're using the returns of createServer() or attachRouting()
+// For using await on the top level CJS, wrap it in async IIFE:
+// (async () => { ... })();
+const { app, httpServer } = await createServer(config, routing);
+
+// Adjust your tests:
+const { responseMock } = await testEndpoint({
+  endpoint,
+  mockFn: jest.fn, // or vi.fn from vitest, required
```

Comment:
it can be confused by `jest.mock`, should find a better naming for it,
probably `fnMethod`